code/game/atoms_movable.dm

```diff
@@ -588,8 +586,6 @@
 			if(moving_diagonally == SECOND_DIAG_STEP)
 				if(!. && set_dir_on_move)
 					setDir(first_step_dir)
-				else if (!inertia_moving)
-					newtonian_move(direct)
```

Comment:
already done by Moved()

Reply:
no, it's not, because moving_diagonally is set during that move, at least for the parallax